polkadot/consensus/src/service.rs

```diff
@@ -265,9 +265,11 @@ impl Service {
 		client: Arc<C>,
 		network: Arc<net::ConsensusService>,
 		transaction_pool: Arc<Mutex<TransactionPool>>,
-		key: ed25519::Pair
+		key: ed25519::Pair,
+		exit: ::exit_future::Exit,
```

Comment:
As I've mentioned in personal discussion I can't say I like this. This just introduced additional complexity to the service interface and leaks away implementation details. What happens if the service is dropped before `exit` is signalled? This is something the user of this interface should not worry about.